### PR TITLE
[s] Moves sensitive data out of GLOB

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

**Comment by optimumtact** [comment]
On what possible planet does this need [s]? It's exploitable by admins only.

File: code/_globalvars/database.dm [hunk 1]
```diff
@@ -1,19 +0,0 @@
-	// MySQL configuration
-
-GLOBAL_VAR_INIT(sqladdress, "localhost")
-GLOBAL_PROTECT(sqladdress)
-GLOBAL_VAR_INIT(sqlport, "3306")
-GLOBAL_PROTECT(sqlport)
-GLOBAL_VAR_INIT(sqlfdbkdb, "test")
-GLOBAL_PROTECT(sqlfdbkdb)
```
Comment (JJRcop) [inline code/_globalvars/database.dm]: I just wanted to comment that this variable name is so cryptic if you don't already know what it is, it's hilarious.
Not asking you to change it, just a silly thing.

**Comment by Chnkr** [comment]
This fixes nothing because you can still VV it

**Comment by optimumtact** [comment]
We can prevent viewvar edits, so surely we can prevent viewvar reads too

**Comment by Cyberboss** [comment]
You can't VV true globals unless the game specifically allows it

**Comment by MrStonedOne** [comment]
no, eric is right, this doesn't fix the issue

you can guess `\ref` id numbers if you know the int order, they are not a security boundary, so somebody could href spoof a link to vv a datum that points to this datum

it would have to be a real global.

**Comment by Cyberboss** [comment]
Wew

**Comment by MrStonedOne** [comment]
>We can prevent viewvar edits, so surely we can prevent viewvar reads too

Not really, vv yes, but so many things use `.vars` on an argument that admin proc call could get past anything we make, and even if we close all those holes up, more will come.

The super secure stuff should just be real globals

in fact, the sql stuff could just be read from a file on the fly, it would make changing the password during headmin term changes easier if i had a way to update it.

**Comment by optimumtact** [comment]
that's extra io overhead on every db connection though

**Comment by MrStonedOne** [comment]
but, oj, there will 99% of the time only be 1 db connection per round.

**Comment by Chnkr** [comment]
Even if you put it into a file there are still ways to read it using admin buttons

**Comment by Cyberboss** [comment]
Please elaborate, because if that's true, that's the biggest possible
security hole we have

**Comment by Chnkr** [comment]
tooltips ;-)

**Comment by Chnkr** [comment]
I'm also pretty sure you can use SDQL2 to read files

File: code/modules/admin/verbs/debug.dm [hunk 2]
```diff
@@ -755,11 +755,11 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
...
 		return
 
-	GLOB.medals_enabled = !GLOB.medals_enabled
+	global.medals_enabled = !global.medals_enabled
 
-	message_admins("<span class='adminnotice'>[key_name_admin(src)] [GLOB.medals_enabled ? "disabled" : "enabled"] the medal hub lockout.</span>")
-	feedback_add_details("admin_verb","Toggle Medal Disable") // If...
-	log_admin("[key_name(src)] [GLOB.medals_enabled ? "disabled" : "enabled"] the medal hub lockout.")
+	message_admins("<span class='adminnotice'>[key_name_admin(src)] [global.medals_enabled ? "disabled" : "enabled"] the medal hub lockout.</span>")
+	feedback_add_details("admin_verb","TMH") // If...
```
Comment (AnturK) [inline code/modules/admin/verbs/debug.dm]: ?

**Pull request merged**